File: app/src/main/java/com/ninhttd/devtest/presentation/product/view/ProductListView.java
package com.ninhttd.devtest.presentation.product.view;

import com.ninhttd.devtest.base.BaseView;
import com.ninhttd.devtest.data.dto.ResponseDTO;
import com.ninhttd.devtest.data.entity.Product;
import com.ninhttd.devtest.data.entity.ProductLevel1;

import java.util.List;

public interface  ProductListView extends BaseView {
     void onLoadDataSucces(ResponseDTO<ProductLevel1> viewDTOResponseListDTO);
     void onLoadDataFailed(Throwable e);

    void appendData(ResponseDTO<ProductLevel1> viewDTOResponseListDTO);

    void showLoadingMore();

    void hideLoadingMore();

    void onSearch(List<Product> products);
}

File: app/src/main/java/com/ninhttd/devtest/presentation/product/view/ProductListView.kt
package com.ninhttd.devtest.presentation.product.view

import com.ninhttd.devtest.base.BaseView
import com.ninhttd.devtest.data.dto.ResponseDTO
import com.ninhttd.devtest.data.entity.ProductEntity
import com.ninhttd.devtest.data.entity.ProductLevel1

interface ProductListView : BaseView {
    fun onLoadDataSucces(viewDTOResponseListDTO: ResponseDTO<ProductLevel1>)

    fun appendData(viewDTOResponseListDTO: ResponseDTO<ProductLevel1>)

    fun showLoadingMore()

    fun hideLoadingMore()

    fun onSearch(productEntities: List<ProductEntity>)
}
